### ISSUE TITLE
Gnome 42 support

### ISSUE THREAD

**Issue by lakotamm** [axxapy/gnome-ui-tune#17]
Hello
I am testing the extension on Gnome 42 and it seems to run fine. Could the version 42 be added into the support list?

Thank you!

**Comment by axxapy** [comment]
Experimental support is in review: https://extensions.gnome.org/review/30041, will be available soon.
Alternatively, can be installed from releases page: https://github.com/axxapy/gnome-ui-tune/releases

Making this issue dedicated to Gnome42, will close others

**Comment by chndzcl** [comment]
![image](https://user-images.githubusercontent.com/15052784/164965460-23db4c2b-ba0b-4625-91f1-e4632c44ce91.png)

@axxapy this is a screenshot of my system (ubuntu 22.04), i am not sure whether this is related to #18 ?

**Comment by jacobmischka** [comment]
Same as above, the previews are tiny again on non-primary monitors in Gnome 42 (essentially the same as #2).

**Comment by axxapy** [comment]
@chndzcl @jacobmischka Thank you folks for reports. 
I believe ca2b63b fixed this problem. 
Update is already live, gnome should offer you to update extension. (do not forget to restart gnome-shell after updating)

Waiting for your feedback, if issue is resolved, and will close this ticket as gnome-42 support seems to be mature.

**Comment by jeisinge** [comment]
First, thank you for the great work on this extension --- these enhancements should be made default!  After upgrading to Ubuntu 22.04, I am seeing an issue.

# Problem
When I log in, and go to the overview, I do not see the desktop:

![no_desktop](https://user-images.githubusercontent.com/8866793/166106992-ee19ecd1-a8dc-47c7-ab4e-e640db16376a.png)

And, then, after starting a couple of applications, I continue to not see the desktop and not see those windows:
![no_desktop_no_windows](https://user-images.githubusercontent.com/8866793/166107000-277ee88e-871e-4aaa-b3dc-dc30a4e40087.png)

# Work around
I noticed that a workaround is to start Chrome.  After Chrome starts, the desktop and window overviews work again.  Starting other applications like Firefox or Electron apps do not fix this.

# Details
This same issue happens on the previous version of the extensions and the version I updated this morning.  This might be related to the dual monitor issue as well.

I have also had issues with `gjs` - sometimes a window closes very slowly and then I see `gjs` in my task bar with transparent windows with only an X on them in the overview.  Clicking the X does nothing.  Running `killall gjs` fixes this issue, but also kills some features like desktop notifications for Firefox and Chrome.
![gjs_windows](https://user-images.githubusercontent.com/8866793/166107266-40bb4937-608f-4812-8c39-ba5b226d4fea.png)

In my Gnome Extensions, I noticed the DING extension runs with `gjs` and has an update.  But, since DING is an Ubuntu package, I cannot update via the Exentions Manager. 
![image](https://user-images.githubusercontent.com/8866793/166107211-728fca78-ed97-4730-9e7f-2010e648ffec.png)

<pre>$ ps aux | grep gjs
jacob      34486  0.0  0.1 2873836 27836 ?       Sl   08:05   0:00 /usr/bin/<font color="#C01C28"><b>gjs</b></font> /usr/share/gnome-shell/org.gnome.Shell.Notifications
jacob      34814  0.0  0.1 2931316 27760 ?       Sl   08:05   0:00 /usr/bin/<font color="#C01C28"><b>gjs</b></font> /usr/share/gnome-shell/org.gnome.ScreenSaver
jacob      34912  0.1  0.4 3055804 72664 ?       Sl   08:05   0:01 <font color="#C01C28"><b>gjs</b></font> /usr/share/gnome-shell/extensions/ding@rastersoft.com/ding.js -E -P /usr/share/gnome-shell/extensions/ding@rastersoft.com -M 1 -D 1920:0:1920:1080:1:0:0:0:0:0 -D 0:0:1920:1080:1:27:0:74:0:1
jacob      38988  0.0  0.0   9212  2364 pts/0    S+   08:20   0:00 grep --color=auto <font color="#C01C28"><b>gjs</b></font>
</pre>

**Comment by Ethran** [comment]
> When I log in, and go to the overview, I do not see the desktop:

I have the same issue, I think it might be related to the Ubuntu dock extension. I have found out that turning off option  "Hide Search"  repairs it. If only "Hide Search" is enabled problem appears.
![Screenshot from 2022-04-30 20-51-34](https://user-images.githubusercontent.com/77555700/166119036-511a7896-8110-44e0-b27f-b07303652711.png)

**Comment by axxapy** [comment]
@jeisinge @Ethran 
I took ubuntu 22.04 iso and run it in new virtual machine. Latest version of extension (released today) works without any issues.
Most likely these issues were caused by the latest update which used too new syntax for gjs version shipped with ubuntu.

Try latest version, should work fine.

If issue still persists, try to disable all extensions, enable only this one and enabling extensions one-by-one to find out which one may conflict with my extension

**Comment by Ethran** [comment]
unfortunately, bug still appears:
I created a new user account, and installed only this extension using extension manager (installed via flatpack), log out, log in, and the same result. If I turn off every option in extension settings and leave only hide search enable, the bug still appears. With all options disabled everything works as intended.
I got the same results logging in the Wayland session and x11 session.
Going to overview and writing something fix this bug until next login. I did not find any other way to "repair" it.

PS: I have upgraded from Ubuntu 21.10 to 22.04 lts - this is not freshly installed system, I will try to test it with fresh install.

**Comment by axxapy** [comment]
@Ethran what do you see in your journal?

**Comment by barreiroleo** [comment]
I have the same issue on Arch, disabling 'hide-search' the problem is gone. Actual Gnome version is 42.1. I have to test the git version. Chears!

**Comment by jacobmischka** [comment]
> I believe ca2b63b9a36a369a2e617ac2d5e1949818ec8180 fixed this problem.

Can confirm that the smaller workspaces on non-primary monitors are fixed now, thanks!

For what it's worth, I'm also on Arch, and everything seems to be perfect for me now.

**Comment by Ethran** [comment]
@axxapy
I don't know what pieces of information are related to this bug so here are all logs for test user - power on, log in, and go to overview. I hope it'll be helpful. 
[logs.txt](https://github.com/axxapy/gnome-ui-tune/files/8602361/logs.txt)

**Comment by chndzcl** [comment]
> @chndzcl @jacobmischka Thank you folks for reports. I believe [ca2b63b](https://github.com/axxapy/gnome-ui-tune/commit/ca2b63b9a36a369a2e617ac2d5e1949818ec8180) fixed this problem. Update is already live, gnome should offer you to update extension. (do not forget to restart gnome-shell after updating)
> 
> Waiting for your feedback, if issue is resolved, and will close this ticket as gnome-42 support seems to be mature.


@axxapy Many thanks for your work, my issue is resolved now.

**Comment by jeisinge** [comment]
The window overview is still not showing up initially.  However, typing anything is a simple, easy workaround!

If you would like for me to capture logs, let me know the specific logs you would like!

**Comment by axxapy** [comment]
> @axxapy I don't know what pieces of information are related to this bug so here are all logs for test user - power on, log in, and go to overview. I hope it'll be helpful. [logs.txt](https://github.com/axxapy/gnome-ui-tune/files/8602361/logs.txt)

There are no errors in your logs related to this extension. But you have a lot of failures from gnome-shell `Can't update stage views actor`

You can see them by filtering your logs `cat logs.txt | grep gnome-shell`

Unfortunately there is not much I can do with these logs. 

Recommendations I can think about:
1. disable all extensions and check logs. If errors are still there, report them to ubuntu bugtracker (launchpad).
2. Try previous versions of this extensions. You can download them on [releases page](https://github.com/axxapy/gnome-ui-tune/releases)

@jeisinge you could try the same ^^
You could also look into logs emitted by gnome-shell with the following command: `journalctl -b -xe /usr/bin/gnome-shell`

**Comment by jeisinge** [comment]
It is still happening for me.

Thanks for the tip on grabbing the logs.  I am attaching the logs after restarting my computer and going to the overview.

[gnome-shell-err.txt](https://github.com/axxapy/gnome-ui-tune/files/8648156/gnome-shell-err.txt)

**Comment by axxapy** [comment]
> It is still happening for me.

you see these logs even when all extensions (including this one) are disabled?

**Comment by jeisinge** [comment]
Nope.  After I disabled extensions and restarted my computer, no logs happened when I opened the overview.  Overview showed all windows.

Attached you will find the full logs of the most recent attempt.  I don't think I captured everything the first time.  Line 72 is when we have additional output when extensions are enabled.

[open-overview-extensions.txt](https://github.com/axxapy/gnome-ui-tune/files/8656305/open-overview-extensions.txt)
